sessions/src/test/kotlin/io/spine/examples/pingh/sessions/SessionsContextSpec.kt

```diff
@@ -0,0 +1,148 @@
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.sessions
+
+import io.kotest.matchers.shouldBe
+import io.spine.base.EventMessage
+import io.spine.examples.pingh.sessions.event.UserLoggedIn
+import io.spine.examples.pingh.sessions.event.UserLoggedOut
+import io.spine.examples.pingh.sessions.given.*
+import io.spine.protobuf.AnyPacker
+import io.spine.server.BoundedContextBuilder
+import io.spine.testing.server.EventSubject
+import io.spine.testing.server.blackbox.ContextAwareTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+
+@DisplayName("Sessions Context should")
+class SessionsContextSpec : ContextAwareTest() {
+
+    override fun contextBuilder(): BoundedContextBuilder =
+        newBuilder()
+
+    @Nested
+    inner class `handle the 'LogUserIn' command` {
+
+        private lateinit var session: SessionId
+
+        @BeforeEach
+        fun sendCommand() {
+            session = createSession()
+            val command = logUserIn(session)
+            context().receivesCommand(command)
+        }
+
+        @Test
+        fun `emit 'UserLoggedIn' event`() {
+            val expected = with(UserLoggedIn.newBuilder()) {
+                id = session
+                build()
+            }
+            val events = assertEvents(UserLoggedIn::class.java)
+            events.hasSize(1)
+            events.message(0)
+                .comparingExpectedFieldsOnly()
+                .isEqualTo(expected)
+        }
+
+        @Test
+        fun `update the 'UserSession' entity`() {
+            val expected = userSession(session)
+            context().assertState(session, expected)
+        }
+    }
+
+    @Nested
+    inner class `handle the 'LogUserOut' command` {
+
+        private lateinit var session: SessionId
+
+        @BeforeEach
+        fun sendCommand() {
+            session = createSession()
+            context().receivesCommand(logUserIn(session))
+            context().receivesCommand(logUserOut(session))
+        }
+
+        @Test
+        fun `emit 'UserLoggedOut' event`() {
+            val expected = with(UserLoggedOut.newBuilder()) {
+                id = session
+                vBuild()
+            }
+            val eventSubject = assertEvents(UserLoggedOut::class.java)
+            eventSubject.hasSize(1)
+            val event = eventSubject.actual()[0]
+            val message = AnyPacker.unpack(event.message, UserLoggedOut::class.java)
```

Comment:
I think this whole piece can be simplified. See [this API call](https://github.com/SpineEventEngine/core-java/blob/1.x-dev/testutil-server/src/main/java/io/spine/testing/server/blackbox/BlackBoxContext.java#L678).